pwnlib/context/__init__.py

```diff
@@ -1213,6 +1222,21 @@ def buffer_size(self, size):
         """
         return int(size)
 
+    @_validator
+    def cache_dir_base(self, new_base):
+        """Base directory to use for caching content.
+
+        Changing this to a different value will clear the `cache_dir` path
+        stored in TLS since a new path will need to be generated to respect the
+        new `cache_dir_base` value.
+        """
+
+        if new_base != self.cache_dir_base:
+            del self._tls["cache_dir"]
+        if os.access(new_base, os.F_OK) and not os.access(new_base, W_OK):
```

Comment:
Looks like a `NameError` on `W_OK`
```suggestion
        if os.access(new_base, os.F_OK) and not os.access(new_base, os.W_OK):
```